src/lumigo_tracer/wrappers/http/sync_http_wrappers.py

```diff
@@ -204,9 +205,28 @@ def _requests_wrapper(func, instance, args, kwargs):
     This is the wrapper of the function `requests.request`.
     This function is being wrapped specifically because it initializes the connection by itself and parses the response,
         which creates a gap from the traditional http.client wrapping.
+    Moreover, these "extra" steps may raise exceptions. We should attach the error to the http span.
     """
     start_time = datetime.now()
-    ret_val = func(*args, **kwargs)
+    try:
+        ret_val = func(*args, **kwargs)
+    except Exception as exception:
+        with lumigo_safe_execute("requests wrapper exception occurred"):
+            url = args[1]
+            if HttpState.previous_request and HttpState.previous_request.host in url:
+                span = SpansContainer.get_span().get_last_span()
+            else:
+                span = add_request_event(
+                    HttpRequest(
+                        host=url,
+                        method=args[0].upper(),
+                        uri=url,
+                        body=kwargs.get("data"),
+                        headers=kwargs.get("headers"),
+                    )
+                )
+            SpansContainer.get_span().add_exception_to_span(span, exception, [])
```

Comment:
Won't it add the exception for the entire execution?
I think the error suppose to be added only to the http call.

Reply:
no, it'll add exception to the provided span. This span is th http span (and not the function span)

Reply:
This might be problematic:
- The user might wrap the call in try-except so he might not be interested in this error but we'll send him an alert.
- What happens if there is a real exception in the lambda? Won't it override the error?
- What happens there are multiple http calls that fails? Each one will override the previous one?

Reply:
after the talk changed to static method to avoid future confusion